mailbox/store/src/main/java/org/apache/james/mailbox/store/quota/ListeningCurrentQuotaUpdater.java

```diff
@@ -82,43 +87,50 @@ public void event(Event event) throws MailboxException {
-        long expungedSize = totalSize(expunged);
-        long expungedCount = (long) expunged.getUids().size();
-        // Expunge event can contain no data (expunge performed while no messages marked \Deleted)
-        if (expungedCount != 0 && expungedSize != 0) {
-            currentQuotaManager.decrease(quotaRoot, expungedCount, expungedSize);
-        }
-        eventBus.dispatch(
-            EventFactory.quotaUpdated()
-                .randomEventId()
-                .user(expunged.getUsername())
-                .quotaRoot(quotaRoot)
-                .quotaCount(quotaManager.getMessageQuota(quotaRoot))
-                .quotaSize(quotaManager.getStorageQuota(quotaRoot))
-                .instant(Instant.now())
-                .build(),
-            NO_REGISTRATION_KEYS)
-            .block();
+    private void handleExpungedEvent(Expunged expunged, QuotaRoot quotaRoot) {
+        computeQuotaOperation(expunged, quotaRoot).ifPresent(Throwing.<QuotaOperation>consumer(quotaOperation -> {
+            currentQuotaManager.decrease(quotaOperation);
+
+            eventBus.dispatch(
+                EventFactory.quotaUpdated()
+                    .randomEventId()
+                    .user(expunged.getUsername())
+                    .quotaRoot(quotaRoot)
+                    .quotaCount(quotaManager.getMessageQuota(quotaRoot))
+                    .quotaSize(quotaManager.getStorageQuota(quotaRoot))
+                    .instant(Instant.now())
+                    .build(),
+                NO_REGISTRATION_KEYS)
+                .block();
+        }).sneakyThrow());
+    }
+
+    private void handleAddedEvent(Added added, QuotaRoot quotaRoot) {
+        computeQuotaOperation(added, quotaRoot).ifPresent(Throwing.<QuotaOperation>consumer(quotaOperation -> {
+            currentQuotaManager.increase(quotaOperation);
+
+            eventBus.dispatch(
+                EventFactory.quotaUpdated()
+                    .randomEventId()
+                    .user(added.getUsername())
+                    .quotaRoot(quotaRoot)
+                    .quotaCount(quotaManager.getMessageQuota(quotaRoot))
+                    .quotaSize(quotaManager.getStorageQuota(quotaRoot))
+                    .instant(Instant.now())
+                    .build(),
+                NO_REGISTRATION_KEYS)
+                .block();
+        }).sneakyThrow());
     }
 
-    private void handleAddedEvent(Added added, QuotaRoot quotaRoot) throws MailboxException {
-        long addedSize = totalSize(added);
-        long addedCount = (long) added.getUids().size();
-        if (addedCount != 0 && addedSize != 0) {
-            currentQuotaManager.increase(quotaRoot, addedCount, addedSize);
+    private Optional<QuotaOperation> computeQuotaOperation(MetaDataHoldingEvent metaDataHoldingEvent, QuotaRoot quotaRoot) {
+        long size = totalSize(metaDataHoldingEvent);
+        long count = (long) metaDataHoldingEvent.getUids().size();
```

Comment:
please avoid raw cast as it defeats all compiler checks. You can use `Integer.toUnsignedLong` instead